### PR TITLE
PAL-258 Logging improvements to palisade service

### PR DESCRIPTION
**Pull request by dev930018**
As this may be the first of many similar, it is worth ensuring the initial set of changes to the palisade service are complete

The ticket suggests:
> 1) Log at INFO level for receipt of a message
> 2) Log at INFO level for a message sent to another service
> 3) Log at DEBUG for individual function calls with more metadata
> 4) In the event of ERRORs include the METAdata of what was being handled

If this is the full set of cases for logging, we should only see messages on INFO if services are communicating with one another

### CODE REVIEW

File: palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/service/AuditServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    private AuditService auditService;
+
+    @Mock
+    AuditClient auditClient;
+    @Mock
+    Executor executor;
+
+    @Before
+    public void setUp() {
+        logger = (Logger) LoggerFactory.getLogger(AuditService.class);
```
Comment (m09526) [inline palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/service/AuditServiceTest.java:55]: Unecessary cast

**Review by m09526 — CHANGES_REQUESTED** [review]

File: palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/service/AuditServiceTest.java (hunk 1 above)
Comment (dev930018) [reply palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/service/AuditServiceTest.java:55]: The Logback Loggers allow us to add appenders and listen to logging statements, but not in slf4j

This is more verbosely ch.qos.logback.Logger logger = org.slf4j.LoggerFactory.getLogger(~.class);

Comment (dev930018) [reply palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/service/AuditServiceTest.java:55]: The name is a bit misleading, but the cast is needed

**Review by m09526 — DISMISSED** [review]

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/config/ApplicationConfiguration.java [hunk 2]
```diff
@@ -155,6 +161,7 @@ public ObjectMapper objectMapper() {
     @Bean(name = "eureka-client")
     @ConditionalOnProperty(prefix = "eureka.client", name = "enabled")
     public ServiceInstanceRestController eurekaClient() {
+        LOGGER.debug("Instantiated eurekaClient");
```
Comment (dev-958) [inline palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/config/ApplicationConfiguration.java]: INFO please, we would always want to see this at startup

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/config/ApplicationConfiguration.java [hunk 3]
```diff
@@ -164,7 +171,7 @@ public Executor getAsyncExecutor() {
         return Optional.of(new ThreadPoolTaskExecutor()).stream().peek(ex -> {
             ex.setThreadNamePrefix("AppThreadPool-");
             ex.setCorePoolSize(6);
-            LOGGER.info("Starting ThreadPoolTaskExecutor with core = [{}] max = [{}]", ex.getCorePoolSize(), ex.getMaxPoolSize());
+            LOGGER.debug("Starting ThreadPoolTaskExecutor with core = [{}] max = [{}]", ex.getCorePoolSize(), ex.getMaxPoolSize());
```
Comment (dev-958) [inline palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/config/ApplicationConfiguration.java]: ditto

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/ResultAggregationService.java [hunk 4]
```diff
@@ -119,7 +123,7 @@ private void auditRequestReceivedException(final RegisterDataRequest request, fi
                         .withContext(request.getContext())
                         .withException(ex)
                         .withServiceClass(serviceClass);
-        LOGGER.debug("Error handling: " + ex.getMessage());
+        LOGGER.debug("Auditing request exception: \n\t{}\n\t{}", request, ex);
```
Comment (dev-958) [inline palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/ResultAggregationService.java]: Here we are logging an exception event, we will only be here if an exception has occurred so we will certainly want to see that meta data whatever the level so ERROR level is applicable.

**Review by dev-958 — CHANGES_REQUESTED** [review]
see inline comments

**Review by m09526 — APPROVED** [review]

**Review by dev-958 — APPROVED** [review]

**Pull request merged**